openedx/core/djangoapps/discussions/models.py

```diff
@@ -12,9 +13,82 @@
 from simple_history.models import HistoricalRecords
 
 from lti_consumer.models import LtiConfiguration
+from organizations.models import Organization
 
+from openedx.core.djangoapps.config_model_utils.models import StackedConfigurationModel
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
+log = logging.getLogger(__name__)
+
+
+class ProviderFilter(StackedConfigurationModel):
```

Comment:
Should this perhaps be DiscussionProviderFilter? There may be other kinds of providers in the future. 

Additionally, is there any potential confusion here about the inheritance of fields? 

i.e. If someone adds a site-level deny for 'lti-provider1' and someone else adds an org-level allow for 'lti-provider1' the org-level as an exception, the site level deny will override the org-level allow. 

Perhaps this should be a single filter, just an allow-list so that anything added there will show up. That way an org-level filter will be able to override a site level filter etc.

Reply:
> Should this perhaps be DiscussionProviderFilter? There may be other kinds of providers in the future.

My thought was that since this is inside the `discussions` django app, it's already sufficiently namespaced, eg: `discussions.models.ProviderFilter`.

This also gives us less verbose database tables, `discussions__provider_filter` instead of `discussion__discussions_provider_filter`.

I know we don't do this everywhere, so we can chose to be more verbose; I just prefer the brevity :)

Reply:
> Additionally, is there any potential confusion here about the inheritance of fields?

There shouldn't be any inheritance.

If config is set for a course, that is used exclusively.
Else if config is used for the Org, that is used exclusively.
If no other config is used, the global defaults are used.

Reply:
> Perhaps this should be a single filter, just an allow-list so that anything added there will show up. That way an org-level filter will be able to override a site level filter etc.

This downside to this approach is that we risk "hiding" new providers;
eg: if you just want to hide a single provider, you'd have to set an allow-list of all other providers. If we add new available providers after this allowlist is created, they'll never show up because the allowlist doesn't know about them.

Reply:
I miswrote. I meant stacking of fields not inheritence. 
i.e. If at the org level you allow 'piaza,yellowdig' and someone adds at the course level 'providerx' then at the course level the allow list will only be 'providerx' AFAIK, but some people might expect it to be 'piazza,yellowdig,providerx'.

Reply:
I think I agree with you @xitij2000 that that is a reasonable assumption, but I think given this is a Django admin setting, we can provide very clear help text in the admin setting, and this is not learner-facing (specifically this is for course set-up, which is not as time sensitive as, say, an exam), I am ok with a simpler model that doesn't take great pains to consider when and when not to inherit settings.